src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,373 @@
+        {
+            // Copy all DEB packages as well
+            foreach (string debPackage in Directory.GetFiles(Config.AssetsDirectory, "*.deb", SearchOption.AllDirectories))
+            {
+                File.Copy(debPackage, Path.Combine(_contextDir, Path.GetFileName(debPackage)));
+            }
+
+            // Download NetStandard 2.1 packages
+            DownloadFileAsync(NetStandard21DebPackage, Path.Combine(_contextDir, Path.GetFileName(NetStandard21DebPackage))).Wait();
+            _debContextInitialized = true;
+        }
+
+        if (!_sharedContextInitialized)
+        {
+            // Copy nuget packages
+            string nugetPackagesDir = Path.Combine(_contextDir, "packages");
+            Directory.CreateDirectory(nugetPackagesDir);
+            foreach (string package in Directory.GetFiles(Config.PackagesDirectory, "*.nupkg", SearchOption.AllDirectories))
+            {
+                File.Copy(package, Path.Combine(nugetPackagesDir, Path.GetFileName(package)));
+            }
+
+            // Copy and update NuGet.config from scenario-tests repo
+            string newNuGetConfig = Path.Combine(_contextDir, "NuGet.config");
+            File.Copy(Config.ScenarioTestsNuGetConfigPath, newNuGetConfig);
+            InsertLocalPackagesPathToNuGetConfig(newNuGetConfig, "/packages");
+
+            // Find the scenario-tests package and unpack it to the context dir, subfolder "scenario-tests"
+            string? scenarioTestsPackage = Directory.GetFiles(nugetPackagesDir, "Microsoft.DotNet.ScenarioTests.SdkTemplateTests*.nupkg", SearchOption.AllDirectories).FirstOrDefault();
+            if (scenarioTestsPackage == null)
+            {
+                Assert.Fail("Scenario tests package not found");
+            }
+
+            ZipFile.ExtractToDirectory(scenarioTestsPackage, Path.Combine(_contextDir, "scenario-tests"));
+            _sharedContextInitialized = true;
+        }
+    }
+
+    private void InsertLocalPackagesPathToNuGetConfig(string nuGetConfig, string localPackagesPath)
+    {
+        XDocument doc = XDocument.Load(nuGetConfig);
+        if (doc.Root != null)
+        {
+            XElement? packageSourcesElement = doc.Root.Element("packageSources");
+            if (packageSourcesElement != null)
+            {
+                XElement? clearElement = packageSourcesElement.Element("clear");
+                if (clearElement != null)
+                {
+                    XElement newAddElement = new XElement("add",
+                    new XAttribute("key", "local-packages"),
+                        new XAttribute("value", localPackagesPath));
+
+                    clearElement.AddAfterSelf(newAddElement);
+                }
+            }
+
+            doc.Save(nuGetConfig);
+        }
+    }
+
+    private void DistroTest(string baseImage, PackageType packageType)
+    {
+        // Order of installation is important as we do not want to use "--nodeps"
```

Comment:
Consider factoring out the logic to define the package list into a separate method.  IMO it hurts the readability of the method.

Reply:
Yes, this is something I considered earlier, makes sense to do it now.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/7dd918feef4c476fe34ed8ec3c51da831c3ad621